include/hipSYCL/common/filesystem.hpp

```diff
@@ -36,22 +36,24 @@ std::string
 join_path(const std::string &base,
           const std::vector<std::string> &additional_components);
 
-std::vector<std::string> list_regular_files(const std::string &directory);
 std::vector<std::string> list_regular_files(const std::string &directory,
-                                            const std::string &extension);
+                                            std::error_code &EC);
+std::vector<std::string> list_regular_files(const std::string &directory,
+                                            const std::string &extension,
+                                            std::error_code &EC);
 
 /// Writes data atomically to filename
 bool atomic_write(const std::string& filename, const std::string& data);
 
 /// Removes a file, returns true if successful.
 bool remove(const std::string &filename);
 
+// Replaces runtime specialized variables, such as $HIPSYCL_PATH in the path.
+std::string replace_known_variables(const std::string path);
```

Comment:
Why is this necessary? the replacement that is done there is likely to support a different set of substitutions than what `acpp` supports, and I'd like to avoid having to maintain such an engine twice.
This seems to be used mainly for the clang path - maybe we just add a function that specifically returns the clang path?